qml/TabLayout.qml

```diff
@@ -248,37 +257,149 @@ Item {
+                            initialize_domain_view_(stack_id, entity_id, domain_id)
+                        }
+                    }
+                }
             }
         }
     }
 
-    Component {
-        id: domainViewLayout
+    Dialog {
+        id: domain_id_dialog
+
+        property bool enable_ok_button: false       // disable OK button until user selects domain id
 
-        Rectangle{
-            Text{
-                text: "Here would be the domain view"
+        x: (parent.width - width) / 2
+        y: (parent.height - height) / 2
+
+        width: 300
+
+        modal: true
+        title: "Select DDS Domain"
+
+        footer: DialogButtonBox {
+            id: buttons
+            standardButtons: Dialog.Ok | Dialog.Cancel
+        }
+
+        Component.onCompleted: {
+            controller.update_available_entity_ids("Domain", "getDataDialogSourceEntityId")
+        }
+
+        onAboutToShow: {
+            custom_combobox.currentIndex = -1
+            controller.update_available_entity_ids("Domain", "getDataDialogSourceEntityId")
+            custom_combobox.recalculateWidth()
+            enable_ok_button = false
+            buttons.standardButton(Dialog.Ok).enabled = false
+        }
+
+        onEnable_ok_buttonChanged: {
+            buttons.standardButton(Dialog.Ok).enabled = domain_id_dialog.enable_ok_button
+        }
+
+        AdaptiveComboBox {
+            id: custom_combobox
+            textRole: "name"
+            valueRole: "id"
+            displayText: currentIndex === -1
+                            ? ("Please choose a Domain ID")
+                            : ("DDS Domain " + currentText)
+            model: entityModelFirst
+
+            Component.onCompleted:
+            {
+                controller.update_available_entity_ids("Domain", "getDataDialogSourceEntityId")
+                currentIndex = -1
+                custom_combobox.recalculateWidth()
             }
+
+            onActivated: {
+                domain_id_dialog.enable_ok_button = true
+                custom_combobox.recalculateWidth()
+            }
+        }
+
+        onAccepted:
+        {
+            open_domain_view_(
+                tabLayout.tab_model_[current_]["stack_id"],
+                entityModelFirst.get(custom_combobox.currentIndex).id,
+                entityModelFirst.get(custom_combobox.currentIndex).name)
         }
     }
 
     function create_new_tab()
     {
         var idx = tabLayout.tab_model_.length
         tabLayout.tab_model_[idx] = {"idx" : idx, "title": "New Tab", "stack_id":last_index_}
-        var new_stack = stack_component.createObject(null, {"id": last_index_, "anchors.fill": "parent"})
+        var new_stack = stack_component.createObject(null, {"stack_id": tabLayout.tab_model_[idx]["stack_id"],
+            "anchors.fill": "parent"})
         last_index_++
         stack_layout.children.push(new_stack)
         refresh_layout(idx)
```

Comment:
If `idx` doesn't match `currentIndex`, `stack_layout.currentIndex` update is already managed in `refresh_layout()` call.

Reply:
After testing, I've realized that this call was forced to be there to ensure that closing your current tab (and current is the last tab) it also changes the stack layout. If that call is removed, UI does not refresh properly the stack view.